File: nim/darts/darts.nim
import std/math
  
proc distanceFromCentre(x, y: float): float =
  hypot(x, y)
  
proc score*(x, y: float): int =
  let dist = distanceFromCentre(x, y);

  if dist <= 1.0:  return 10
  if dist <= 5.0:  return 5
  if dist <= 10.0: return 1
  
  return 0

File: haskell/darts/Iteration1/src/Darts.hs
module Darts (score) where

dist :: Float -> Float -> Float
dist x y = sqrt ((x ^ 2) + (y ^ 2))

score :: Float -> Float -> Int
score x y
  | d > 10 = 0
  | d > 5 = 1
  | d > 1 = 5
  | d >= 0 = 10
  where
    d = dist x y
